### PR TITLE
Fix orderBy tests on functional get dapps - Closes #977

### LINKED ISSUE #977: Fix intermittent test failures related to data sorting
**Expected behaviour** 
Sorting of data returned by database must match with the sorting to native javascript sort method. 

**Actual Behaviour** 
PostgreSQL sort data case insensitive while javascript sort lists by case sensitive manner. So while matching sorted result sets in test cases, they get failed. Since tests rely on randomly generated data, so this behaviour triggered only if data contains some capital or special character.

### PR DESCRIPTION
**Pull request by diego-G**
Closes #977

### CODE REVIEW

**Comment by nazarhussain** [comment]
@diego-G @karmacoma In my last PR I converted `sort` to a new method `dbSort`. So I don't think changing it back will actually solve the issue. I have the data set for which `sort` is failing as well the dataset for which `dbSort` is failing. None of the solution is perfect, but dbSort function solved the special characters issue very well. 

Since we change `sort` to `dbSort` all over tests, so what ever technique we need to implement we should do it inside `dbSort` and don't touch the tests.

**Comment by diego-G** [comment]
I've study the case deeply. Javascript sort function order case sensitive by default. Our Jenkins Database has a database configuration which orders in case insensitive mode. That is the problem. Now we can see the error always reproducing in the server because I am creating 20 apps instead of 2. Then there are many probabilities to encounter the scenario.

Locally in my machine, postgres `order by` is always case sensitive. Either having Collate as `C`, or as `en_US.UTF-8`. Then tests pass always.
In Jenkins, postgres order by is case sensitive having Collate as `C`,  case insensitive having `en_US.UTF-8`. The second is the one we have configured. Then test don't pass.

File: test/unit/logic/account.js [hunk 1]
```diff
@@ -376,36 +382,34 @@ describe('account', function () {
...
+				expect(res).to.eql(_.sortBy(res, 'username'));
 				done();
 			});
 		});
 
 		it('should sort the result according to field type in descending order', function (done) {
 			account.getAll({sort: 'username:desc'}, ['username'], function (err, res) {
 				expect(err).to.not.exist;
-				expect(_(res).map('username').dbSort('desc')).to.eql(_.map(res, 'username'));
+				expect(res).to.eql(_.sortBy(res, 'username').reverse());
```
Comment (nazarhussain) [inline test/unit/logic/account.js]: @karmacoma We recently change all occurrence of `sort` and `sortBy` and `orderBy` occurrences in all test cases to one method `dbSort`. So we should use same sorting pattern everywhere in test cases. I don' think reverting it back in one test file is a good idea.

**Comment by nazarhussain** [comment]
@diego-G Related to your comment above, As per my experience PostgreSQL installation on Linux pickup the system collate as default. An system collate is dependent on the language your select during the installation. I never encountered a case where Linux system don't have some standard collate. Logically collate "C" refers to a custom collate where you don't want to choose between default locales available during installation. So I don't think this scenario is going to occurs on any node/peer running Linux to have "C" collate, unless some one crafted the linux distro himself. 

Related to Mac, I believe its the bug, that what ever collate your system have, PostgreSQL always say case sensitive. 

To move to a conclusion on this topic, we already decided as per slack channel that we will stick to case insensitive sorting. So any fix/solution in our code must focus this. While JS `sort` method is follow case sensitive sorting, so we should not use it.

**Comment by diego-G** [comment]
The solution is really simple: we enforce the **default locale** in lisk-build to built a database matching the **default** behaviour of Javascript Sorting which is **case sensitive**.
Another points to highlight are:
- ASCII sorting is default for M2M
- Since delegate and application names distinguish the case, so should we on the sorting.

**Comment by diego-G** [comment]
@fchavant came out with an even easier workaround. I add it on the last commit. Now we don't need to open an issue in `lisk-build`.

File: test/unit/logic/account.js (hunk 1 above)
Comment (karmacoma) [reply test/unit/logic/account.js]: @nazarhussain I have reverted your PR to include only @diego-G changes.

**Pull request merged**